server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1462,6 +1462,25 @@ final boolean tryRenewSyncCommit() {
         return renewed;
     }
 
+    @Override
+    public boolean shouldFlushToFreeTranslog() {
+        ensureOpen();
+        final long flushThreshold = config().getIndexSettings().getFlushThresholdSize().getBytes();
+        final long uncommittedSizeOfCurrentCommit = translog.uncommittedSizeInBytes();
+        // If flushThreshold is too small, we may continuously flush even there is no uncommitted operations.
+        if (uncommittedSizeOfCurrentCommit < flushThreshold || translog.uncommittedOperations() == 0) {
```

Comment:
maybe put the check `translog.uncommittedOperations() == 0` at the beginning of the `shouldFlushToFreeTranslog` method.